lib/blumquist/errors/invalid_pointer.rb

```diff
@@ -1,8 +1,13 @@
 class Blumquist
   module Errors
     class InvalidPointer < Blumquist::Error
-      def initialize(pointer)
-        super("Could not find pointer #{pointer.to_s.to_json}")
+      def initialize(pointer:, document:)
+          msg = %{
+  Could not find pointer '#{pointer}' in the given document:\n
+  #{JSON.pretty_generate(document)}
+          }
```

Comment:
The indentation might looks wrong, but it is not.  `%{}` take the indentation into account.

Reply:
how about 

```ruby
    msg = "foo " \
                "bar"
```

or 

```ruby
   msg = <<~MSG
     foo bar
MSG
```

Reply:
The second one is equivalent to the one used here. The first one might work better, although I don't think it is an issue to use `%{}` and have the indentation like that.